src/main/java/com/wcc/platform/deserializers/ProgramTypeDeserializer.java

```diff
@@ -0,0 +1,20 @@
+package com.wcc.platform.deserializers;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.wcc.platform.domain.platform.ProgramType;
+import java.io.IOException;
+
+/** Custom deserializer for {@code ProgramType} enum. */
+public class ProgramTypeDeserializer extends JsonDeserializer<ProgramType> {
+
+  @Override
+  public ProgramType deserialize(final JsonParser jsonParser, final DeserializationContext context)
+      throws IOException {
+
+    final var value = jsonParser.getText();
+
+    return ProgramType.fromString(value);
```

Comment:
Jackson default behavior is serializing and deserializing by enum name. I am not sure if there is need for custom class.
https://www.baeldung.com/jackson-serialize-enums

Reply:
This is a custom method: `fromString`  to return the program description: For example MACHINE_LEARNING will be returned `Machine Learning`.